qutebrowser/utils/urlutils.py

```diff
@@ -196,17 +199,21 @@ def fuzzy_url(urlstr, cwd=None, relative=False, do_search=True):
     return url
 
 
-def _has_explicit_scheme(url):
+def _has_explicit_scheme(url, allow_only_scheme=True):
     """Check if an url has an explicit scheme given.
 
     Args:
         url: The URL as QUrl.
+        allow_only_scheme: if set to True the URL is allowed to contain only
+                           the scheme with an empty path.
+
     """
     # Note that generic URI syntax actually would allow a second colon
     # after the scheme delimiter. Since we don't know of any URIs
     # using this and want to support e.g. searching for scoped C++
     # symbols, we treat this as not an URI anyways.
     return (url.isValid() and url.scheme() and
+            (allow_only_scheme or len(url.path()) > 0) and
```

Comment:
I have a few comments about this:

- `len(url.path()) > 0` can just be written as `url.path()` (as empty strings are false)
- Is `allow_only_scheme` really needed? Maybe `foo:` shouldn't be treated as URL in any case (as it makes no sense), not only with multi-line pastes?
- Like this you're treating `http://www.qutebrowser.org` (without trailing slash) as no URL, which is probably wrong. Shouldn't this be `url.host() or url.path()`? (`url.path()` added to interpret e.g. `mailto:foo` as URL).
- Can you add a few unit tests in `tests/unit/utils/test_urlutils.py` for this as well?